ImageChecks.ps1

```diff
@@ -0,0 +1,167 @@
+        if ($extension -eq "raw") {
+            $converted = $Image.split(".")[0]+".vhdx"
+            Write-Host "Converting $Image to $converted..."
+            qemu-img convert -f $extension -O vhdx $Image $converted
+            if ($? -ne "True") {
+                Write-Host "Error while converting"
+                exit
+            }
+            Write-Host "Mounting $converted..."
+            $volume = Mount-VHD -Path $converted -Passthru | Get-Disk | Get-Partition | Get-Volume
+            $volume = $volume.DriveLetter
+            $volume = $volume + ":\"
+            Write-Host "Checking for curtin..."
+            $curtin = ls $volume | Select-String -Quiet "curtin"
+            if ($curtin) {
+                Write-Host "OK"
+            } else {
+                Write-Host "$Image doesn't have curtin"
+            }
+            Dismount-VHD $converted
+            Remove-Item $converted -ErrorAction Stop
+        } else {
+            Write-Host "Wrong format for MAAS"
+            exit
+        }
+    } elseif ($Type -eq "KVM") {
+        if ($extension -eq "raw" -Or $extension -eq "qcow2") {
+            $converted = $Image.split(".")[0]+".vhdx"
+            Write-Host "Converting $Image to $converted..."
+            qemu-img convert -f $extension -O vhdx $Image $converted
+            if ($? -ne "True") {
+                Write-Host "Error while converting"
+                exit
+            }
+            Write-Host "Mounting $converted..."
+            $volume = Mount-VHD -Path $converted -Passthru | Get-Disk | Get-Partition | Get-Volume
+            $volume = $volume.DriveLetter
+            $volume = $volume + ":\"
+            Write-Host "Checking for VirtIO..."
+            $virtio = Get-ChildItem -Recurse -Force $volume -ErrorAction SilentlyContinue `
+                        | Where-Object { ($_.PSIsContainer -eq $false) -and  ( $_.Name -like "vio*") }
+            if ($virtio) {
+                Write-Host "OK"
+            } else {
+                Write-Host "$Image doesn't have VirtIO"
+            }
+            Dismount-VHD $converted
+            Remove-Item $converted -ErrorAction Stop
+        } else {
+            Write-Host "Wrong format for KVM"
+            exit
+        }
+    } else {
+        Write-Host "Wrong type"
+        exit
+    }
+}
+
+function Validate-Image {
+    <#
+    .SYNOPSIS
+    Checks what type of file was given and after that checks its format
+    .DESCRIPTION
+    At first checks to see if the given file must be extracted or not,
+    after that it proceeds with checking its format and
+    in the end it deletes the extracted file if it's the case
+    #>
+    if ($extension -eq "tgz" -or $extension -eq "gz" -or $extension -eq "tar") {
+        $Image = Extract-File $Image
+        $imgExtracted = $Image
+    }
+    $Type = Get-Type $ConfigFile
+    Check-Format $Type $Image
+    if ($imgExtracted) {
+        Delete-Extracted $imgExtracted
+    }
+}
+
+Validate-Image
```

Comment:
use the params with full param names like this:

Validate-Image -<param-name> <param_value> etc
always leave an extra line at the end of the file, so that cat like commands work in a nice way